### ISSUE TITLE
Chef 13 fails due to namespace collisions

### ISSUE THREAD

**Issue by SaltwaterC** [chef-cookbooks/docker#898]
### Cookbook version

12.15.14 due to #897 - however, the code is the same in the latest master

### Chef-client version

13.2.20

### Platform Details

CentOS 7.3

### Scenario:

Install and configure a Docker service for running a Docker registry as container.

### Steps to Reproduce:

```ruby
docker_service 'default' do
  host %w[tcp://127.0.0.1:2376 unix:///var/run/docker.sock]
  tls_verify true
  tls_ca_cert '/etc/nginx/ssl/tls_ca_cert.crt'
  tls_server_cert '/etc/nginx/ssl/tls_server_cert.crt'
  tls_server_key '/etc/nginx/ssl/tls_server_key.key'
  storage_driver 'overlay' if node['test_kitchen']
  action %i[create start]
end
```

### Expected Result:

Successful Chef run.

### Actual Result:

Even though the [namespace collisions](https://docs.chef.io/deprecations_namespace_collisions.html) have been deprecated, these aren't raised as deprecations in Chef 12 which makes the detection of these issues bit more difficult. Judging by the error message, it seems that the properties must be get with `new_resource.property_name` in HWRP's as well, even though the documentation is bit scarce about this.

```bash
         * docker_service[default] action create[2017-07-28T17:09:19+00:00] ERROR: rename install_method to new_resource.install_method (CHEF-19)/tmp/kitchen/cache/cookbooks/docker/libraries/docker_service.rb:56:in `installation'.
       Please see https://docs.chef.io/deprecations_namespace_collisions.html for further details and information on how to correct this problem. at /opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/formatters/doc.rb:419:in `deprecation'


           ================================================================================
           Error executing action `create` on resource 'docker_service[default]'
           ================================================================================

           Chef::Exceptions::DeprecatedFeatureError
           ----------------------------------------
           "rename install_method to new_resource.install_method (CHEF-19)/tmp/kitchen/cache/cookbooks/docker/libraries/docker_service.rb:56:in `installation'.\nPlease see https://docs.chef.io/deprecations_namespace_collisions.html for further details and information on how to correct this problem. at /opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.2.20/lib/chef/formatters/doc.rb:419:in `deprecation'"

           Cookbook Trace:
           ---------------
           /tmp/kitchen/cache/cookbooks/docker/libraries/docker_service.rb:56:in `installation'
           /tmp/kitchen/cache/cookbooks/docker/libraries/docker_service.rb:100:in `block in <class:DockerService>'

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/docker-registry/recipes/default.rb

            63: docker_service 'default' do
            64:   host %w[tcp://127.0.0.1:2376 unix:///var/run/docker.sock]
            65:   tls_verify true
            66:   tls_ca_cert '/etc/nginx/ssl/tls_ca_cert.crt'
            67:   tls_server_cert '/etc/nginx/ssl/tls_server_cert.crt'
            68:   tls_server_key '/etc/nginx/ssl/tls_server_key.key'
            69:   storage_driver 'overlay' if node['test_kitchen']
            70:   action %i[create start]
            71: end
            72:

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/docker-registry/recipes/default.rb:63:in `from_file'

           docker_service("default") do
             action [:create, :start]
             default_guard_interpreter :default
             declared_type :docker_service
             cookbook_name "docker-registry"
             recipe_name "default"
             host ["tcp://127.0.0.1:2376", "unix:///var/run/docker.sock"]
             tls_verify true
             tls_ca_cert "/etc/nginx/ssl/tls_ca_cert.crt"
             tls_server_cert "/etc/nginx/ssl/tls_server_cert.crt"
             tls_server_key "/etc/nginx/ssl/tls_server_key.key"
             storage_driver ["overlay"]
             pidfile "/var/run/docker.pid"
           end

           System Info:
           ------------
           chef_version=13.2.20
           platform=centos
           platform_version=7.3.1611
           ruby=ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
           program_name=chef-solo worker: ppid=207;start=17:07:30;
           executable=/opt/chef/bin/chef-solo
```

**Comment by someara** [comment]
This is fixed up as of 2.15.23. Closing!

**Comment by SaltwaterC** [comment]
Thanks!
